src/com/cloud/chatbot/Cloud.java

```diff
@@ -0,0 +1,196 @@
+            if (Cloud.TODOS.size() <= 0) {
+                Cloud.say("Your TODO list is empty.");
+                break;
+            }
+
+            for (int i = 0; i < Cloud.TODOS.size(); i++) {
+                Todo todo = Cloud.TODOS.get(i);
+                Cloud.say(todo.toString(i + 1));
+            }
+            break;
+        }
+        case MARK: {
+            @Nullable Integer number = Cloud.verifyNumber(manager);
+            if (number == null) {
+                break;
+            }
+
+            Todo todo = Cloud.TODOS.get(number - 1);
+            todo.setComplete(true);
+            Cloud.say(todo.toString(number));
+            break;
+        }
+        case UNMARK: {
+            @Nullable Integer number = Cloud.verifyNumber(manager);
+            if (number == null) {
+                break;
+            }
+
+            Todo todo = Cloud.TODOS.get(number - 1);
+            todo.setComplete(false);
+            Cloud.say(todo.toString(number));
+            break;
+        }
+        case DELETE: {
+            @Nullable Integer number = Cloud.verifyNumber(manager);
+            if (number == null) {
+                break;
+            }
+
+            Todo todo = Cloud.TODOS.remove(number - 1);
+            Cloud.say("Yeeted:");
+            Cloud.say(todo.toString(number));
+            break;
+        }
+        case EXIT: {
+            Cloud.say("\\o");
+            System.exit(0);
+            break;
+        }
+        case UNKNOWN:
+        default: {
+            Cloud.say(
+                String.format(
+                    "\"%s\" is not a valid command.",
+                    command
+                )
+            );
+            break;
+        }
+        }
+    }
+
+    private static @Nullable Todo createTodo(CommandManager manager) {
+        String description;
+        try {
+            description = manager.getDetails();
+        } catch (MissingInputException e) {
+            Cloud.say("Please enter a description for your TODO.");
+            return null;
+        }
+
+        @Nullable FlagManager managerBy = manager.findFlag("by");
+        @Nullable FlagManager managerFrom = manager.findFlag("from");
+        @Nullable FlagManager managerTo = manager.findFlag("to");
+
+        try {
+            if (managerBy != null) {
+                return new Deadline(description, managerBy.getSubInput());
+            }
+            if (managerFrom != null && managerTo != null) {
+                return new Event(description, managerFrom.getSubInput(), managerTo.getSubInput());
+            }
+        } catch (MissingFlagInputException e) {
+            Cloud.say(
+                String.format(
+                    "Please enter a description for the \"%s\" flag.",
+                    e.getFlagText()
+                )
+            );
+            return null;
+        }
+
+        return new Todo(description);
+    }
+
+    private static @Nullable Integer verifyNumber(CommandManager manager) {
```

Comment:
I see what you mean. This one is also more of a utility function. I'm a bit apprehensive about starting static methods with `get` since it sounds to me like an instance getter.

I named it so because part of its logic involves giving error feedback to the user, but at the end of the call, if the number is valid, then the callee needs to actually use said number (while also knowing whether or not it was valid), thus `@Nullable Integer` instead of just `int`.

I am open to better name suggestions.

Reply:
Come to think of it, if the method were meant to return a boolean, I think it would be called something like `isNumberValid()`. That thought occurred to me when reading this part of the code quality guide:

![image](https://github.com/nus-cs2103-AY2324S1/ip/assets/44526554/cde4935d-2609-45ab-b76e-7fb5460de3bd)